src/main/java/com/softserveinc/dokazovi/controller/AuthorController.java

```diff
@@ -56,23 +49,25 @@ public ResponseEntity<AuthorDTO> createAuthor(@Valid @RequestBody AuthorDTO auth
                 .body(authorService.save(author, userPrincipal));
     }
 
-    @PutMapping()
-    @PreAuthorize("hasAuthority('UPDATE_AUTHOR')")
+    @PatchMapping(AUTHOR_UPDATE)
+    @PreAuthorize("hasAuthority('EDIT_AUTHOR')")
     @ApiOperation(value = "update author",
             authorizations = {@Authorization(value = "Authorization")})
     @ApiResponses(value = {
             @ApiResponse(code = 200, message = HttpStatuses.OK, response = AuthorDTO.class),
             @ApiResponse(code = 400, message = HttpStatuses.BAD_REQUEST)
     })
-    public ResponseEntity<AuthorDTO> updateAuthor(@RequestBody AuthorDTO author,
+    public ResponseEntity<AuthorDTO> updateAuthor(
+            @PathVariable("doctorId") Integer doctorId,
+            @RequestBody AuthorDTO author,
```

Comment:
1. For `@PatchMapping(AUTHOR_UPDATE)` you are requesting the same entity as for `@PostMapping()`. I mean `@RequestBody AuthorDTO`
I would suggest to keep `ID` in `AuthorDTO` entity and check if `id` exists - then this is update operation. If `id` is null, then this is a new user creation.

2. PatchMapping is great for partially data changes as it's done for `PostController.java -> setPublishedAt() method`
![image](https://user-images.githubusercontent.com/86650723/192051369-6bad81bc-6d9d-4c8b-bd1b-b9ba3458aba9.png)
I'm not sure, it's appropriate in this case. It's better to use POST or PUT mappings when you are updating the whole author object.